### PR TITLE
Make the Dockerfile build glide-aware.

### LINKED ISSUE #2: Docker image build broken for hostpath example.
It fails with go-wrapper download errors, because the Dockerfile is not glide-aware.

### PR DESCRIPTION
**Pull request by dlorenc**
This should fix #2

### CODE REVIEW

**Comment by dlorenc** [comment]
It might also be better to compile this statically and use a Makefile so the image can be built FROM SCRATCH. I can handle that too if you'd like.

File: docs/demo/hostpath-provisioner/Dockerfile [hunk 1]
```diff
@@ -12,4 +12,9 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM golang:1.7.4-onbuild
+FROM golang:1.7.4
+COPY . /go/src/app
+WORKDIR /go/src/app
+RUN go get github.com/Masterminds/glide
+RUN glide install
```
Comment (wongma7) [inline docs/demo/hostpath-provisioner/Dockerfile]: does it work without '-v'? to remove nested vendor dirs

**Comment by wongma7** [comment]
also can you update this https://github.com/kubernetes-incubator/external-storage/tree/master/docs/demo/hostpath-provisioner#building-and-running-our-hostpath-dynamic-provisioner so the Dockerfile block looks the same. 

I think FROM SCRATCH could be better BUT I just don't want the hostpath-provisioner binary thing (which is huge) lying around. But to be fair I think the containerized build is an even-huger Docker image soo...if you want to handle a FROM SCRATCH build as well I will accept it.

File: docs/demo/hostpath-provisioner/Dockerfile (hunk 1 above)
Comment (dlorenc) [inline docs/demo/hostpath-provisioner/Dockerfile]: It worked with and without it. I added the -v.

**Comment by dlorenc** [comment]
> also can you update this https://github.com/kubernetes-incubator/external-storage/tree/master/docs/demo/hostpath-provisioner#building-and-running-our-hostpath-dynamic-provisioner so the Dockerfile block looks the same.
> 
Done.
> I think FROM SCRATCH could be better BUT I just don't want the hostpath-provisioner binary thing (which is huge) lying around. But to be fair I think the containerized build is an even-huger Docker image soo...if you want to handle a FROM SCRATCH build as well I will accept it.
Cool. I can add that in another PR.

**Comment by wongma7** [comment]
ok Lgtm

**Pull request merged**